### PR TITLE
feat: add closest locations retry to create database

### PR DESCRIPTION
**Pull request by gris**
closes https://github.com/chiselstrike/turso-cli/issues/518

Also refactored the file `internal/turso/databases.go` to stop using `region`

### CODE REVIEW

File: internal/cmd/db_create.go [hunk 1]
```diff
@@ -170,14 +172,49 @@ var createCmd = &cobra.Command{
...
-		regionText := fmt.Sprintf("%s (%s)", locationDescription(client, region), region)
+		locationText := fmt.Sprintf("%s (%s)", locationDescription(client, locationId), locationId)
 
 		start := time.Now()
-		description := fmt.Sprintf("Creating database %s%s in %s ", internal.Emph(name), dbText, internal.Emph(regionText))
+		description := fmt.Sprintf("Creating database %s%s in %s", internal.Emph(name), dbText, internal.Emph(locationText))
 		spinner := prompt.Spinner(description)
 		defer spinner.Stop()
+		_, err = client.Databases.Create(name, locationId, image, extensions)
+		if err != nil && err.Error() == "location error" {
```
Comment (gris) [inline internal/cmd/db_create.go:182]: Refactor error

File: internal/cmd/db_create.go [hunk 2]
```diff
@@ -170,14 +172,49 @@ var createCmd = &cobra.Command{
...
+			spinner.Stop()
+			fmt.Printf("Location %s is currently experiencing issues. Please pick another from the 3 closest locations to %s or try again later\n", internal.Emph(locationId), internal.Emph(locationId))
```
Comment (gris) [inline internal/cmd/db_create.go:184]: Move retry logic to instance creation

File: internal/cmd/db_create.go [hunk 3]
```diff
@@ -170,14 +172,49 @@ var createCmd = &cobra.Command{
...
+
+			locationText = fmt.Sprintf("%s (%s)", locationDescription(client, locationId), locationId)
+
+			description = fmt.Sprintf("Creating database %s%s in %s ", internal.Emph(name), dbText, internal.Emph(locationText))
+			spinner = prompt.Spinner(description)
+			defer spinner.Stop()
+			_, err = client.Databases.Create(name, locationId, image, extensions)
 
-		if _, err := client.Databases.Create(name, region, image, extensions); err != nil {
+			if err != nil {
```
Comment (gris) [inline internal/cmd/db_create.go:213]: Single line err handling

File: internal/cmd/db_create.go [hunk 4]
```diff
@@ -170,14 +172,49 @@ var createCmd = &cobra.Command{
...
+				return fmt.Errorf("please retry later, location %s is also experiencing issues: %w", locationId, err)
```
Comment (gris) [inline internal/cmd/db_create.go:214]: Change wording to something like: We couldn't create your db

File: internal/turso/databases.go [hunk 5]
```diff
@@ -18,6 +18,14 @@ type Database struct {
 
 type DatabasesClient client
 
+type CreateInstanceLocationError struct {
+	err string
+}
+
+func (e *CreateInstanceLocationError) Error() string {
+	return fmt.Sprint(e.err)
```
[the remark this answers is not in the document]
Comment (gris) [reply internal/turso/databases.go:26]: This is not printing just formatting. Maybe Would `return e.err` be better?

File: internal/cmd/db_create.go [hunk 6]
```diff
@@ -279,3 +289,36 @@ func getDatabaseName(args []string) (string, error) {
 	}
 	return codename.Generate(rng, 0), nil
 }
+
+func handleInstanceCreationError(client *turso.Client, name, locationId string, dbText string, image string) (*turso.Instance, string, error) {
+	fmt.Printf("We couldn't create your database at %s.\nPlease try again in a few moments, or pick one of the nearby locations we've selected for you\n", internal.Emph(locationId))
```
Comment (athoscouto) [inline internal/cmd/db_create.go:294]: ⛏️
```suggestion
	fmt.Printf("We couldn't create your database at %s.\nPlease try again in a few moments, or pick one of the nearby locations we've selected for you.\n", internal.Emph(locationId))
```

**Review by athoscouto — APPROVED** [review]
Great!
Let's merge this 🚀

**Pull request merged**